examples/demo-rollup/src/lib.rs

```diff
@@ -43,21 +36,19 @@ pub struct HexKey {
 pub fn get_genesis_config<A: BasicAddress>(
     sequencer_da_address: A,
 ) -> GenesisConfig<DefaultContext> {
-    let hex_key: HexKey = serde_json::from_slice(include_bytes!(
+    let hex_key: HexPrivateAndAddress = serde_json::from_slice(include_bytes!(
         "../../test-data/keys/token_deployer_private_key.json"
     ))
     .expect("Broken key data file");
-    let sequencer_private_key = DefaultPrivateKey::from_hex(&hex_key.hex_priv_key).unwrap();
-    assert_eq!(
-        sequencer_private_key.default_address().to_string(),
-        hex_key.address,
-        "Inconsistent key data",
-    );
+    let key_and_address: PrivateKeyAndAddress<DefaultContext> = hex_key
+        .try_into()
+        .expect("Failed to parse sequencer private key and address");
+    assert!(key_and_address.is_matching(), "Inconsistent key data");
```

Comment:
Why do we deserialize an address in the genesis config if we aren't willing to use it to override the "default" one that would be generated from the private key? Why not just use a serialized private key with no address?

Reply:
We can do that, but if there's incorrect address in the "key" file. It can bring a confusion, if somebody wants to use those files.

I've been in this situation. I wanted to play with sending transaction one to another, but in one file address was not updated, so it took me a while to identify that problem is there. 

This extra check makes sure, that our key files are always correct.